docs/workflow/building/coreclr/android.md

```diff
@@ -1,100 +1,172 @@
-After that, you'll need to forward port 1234 from your Android device to your PC:
+#### WSL2
+
+The app can be run on an emulator running on the Windows host.
+1. Install Android Studio on the Windows host (same versions as in [prerequisites](#prerequisite))
+2. In Windows, create and start an emulator
+3. In WSL, swap the `adb` from the Android SDK in WSL2 with that from Windows
+    - `mv $ANDROID_SDK_ROOT/platform-tools/adb $ANDROID_SDK_ROOT/platform-tools/adb-orig`
+    - `ln -s /mnt/<path-to-adb-on-host> $ANDROID_SDK_ROOT/platform-tools/adb`
+4. In WSL, Make xharness use the `adb` corresponding to the Windows host:
+    - `export ADB_EXE_PATH=$ANDROID_SDK_ROOT/platform-tools/adb`
+5. In WSL, run the `make` command as [above](#running-helloandroid-sample-on-an-emulator)
+
+### Building and running functional tests on an emulator
+
+Similarly to the `HelloAndroid` sample, it is possible to build and run a functional test on Android with CoreCLR on an emulator.
+
+To build and run a functional test on Android with CoreCLR, run the following command from `<repo_root>`:
+
 ```
-adb forward tcp:1234 tcp:1234
+./dotnet.sh build -c Release src/tests/FunctionalTests/Android/Device_Emulator/JIT/Android.Device_Emulator.JIT.Test.csproj /p:TargetOS=android /p:TargetArchitecture=arm64 /t:Test /p:RuntimeFlavor=coreclr
 ```
 
-Finally, install lldb on your PC and connect to the debug server running on your Android device:
+NOTE: Similarly to the `HelloAndroid` sample the emulator needs to be up and running.
 
-```
-lldb-3.9
-(lldb) platform select remote-android
-  Platform: remote-android
- Connected: no
-(lldb) platform connect connect://localhost:1234
-  Platform: remote-android
-    Triple: aarch64-*-linux-android
-OS Version: 23.0.0 (3.10.84-perf-gf38969a)
-    Kernel: #1 SMP PREEMPT Fri Sep 16 11:29:29 2016
-  Hostname: localhost
- Connected: yes
-WorkingDir: /data/local/tmp
+### Useful make commands
 
-(lldb) target create coreclr/pal/tests/palsuite/file_io/CopyFileA/test4/paltest_copyfilea_test4
-(lldb) env LD_LIBRARY_PATH=/data/local/tmp/coreclr/lib
-(lldb) run
+For convenience it is possible to run a single make command which builds all required dependencies, the app and runs it:
+```
+make BUILD_CONFIG=<Debug|Release> TARGET_ARCH=<x64|arm64> RUNTIME_FLAVOR=CoreCLR DEPLOY_AND_RUN=true all -C src/mono/sample/Android
 ```
+
+## Debugging the runtime and the sample app
+
+Managed debugging is currently not supported, but we can debug:
+- Java portion of the sample app
+- Native code for the CoreCLR host and the runtime it self
+
+This can be achieved in `Android Studio` via `Profile or Debug APK`.
+
+### Steps
+
+1. Build the runtime and `HelloAndroid` sample app in `Debug` configuration targeting `arm64` target architecture.
+2. Rename the debug symbols file of the runtime library from `libcoreclr.so.dbg` into `libcoreclr.so.so`, the file is located at: `<repo_root>/artifacts/bin/AndroidSampleApp/arm64/Debug/android-arm64/publish/libcoreclr.so.dbg`
```

Comment:
Passing `-keepnativesymbols true` to build.sh is another way, with that the debug symbols are kept in the .so directly instead of being stripped to .so.db

Reply:
Added a note about it, thanks!